### PR TITLE
perf(extension): update hidden and searched feature conditional

### PR DESCRIPTION
**Pull request by pyshx**
This PR provides refactor from https://github.com/reearth/reearth/pull/893.
It introduces usage of equality binary operation with array support to reduce evaluation time of the conditional style expressions.

### CODE REVIEW

File: extension/src/shared/layerContainers/buildingModel.tsx [hunk 1]
```diff
@@ -134,32 +133,32 @@ export const BuildingModelLayerContainer: FC<TilesetContainerProps> = ({
...
     () =>
       searchedFeatures?.onlyShow
         ? {
             conditions: [
-              ...(searchedFeatures?.conditions?.flatMap(([key, values]) =>
-                values.map(
-                  v => [`${variable(key)} === ${numberOrString(v)}`, "true"] as [string, string],
-                ),
-              ) ?? []),
+              ["${searchedFeaturesIds} == ${gml_id}", "true"],
```
Comment (keiya01) [inline extension/src/shared/layerContainers/buildingModel.tsx:154]: `searchedFeatures.conditions` isn't using id, so I think we don't need to use the variable in this case 🙏

File: extension/src/shared/layerContainers/buildingModel.tsx [hunk 2]
```diff
@@ -134,32 +133,32 @@ export const BuildingModelLayerContainer: FC<TilesetContainerProps> = ({
...
-  );
+  const hiddenFeaturesConditions: ConditionsExpression = {
+    conditions: [["${hiddenFeatureIds} == ${gml_id}", "false"]],
+  };
 
   const searchedFeatures = useAtomValue(searchedFeaturesAtom);
+
+  const defines: Record<string, string> = {
+    hiddenFeaturesIds: JSON.stringify(hiddenFeatures),
+  };
```
Comment (keiya01) [inline extension/src/shared/layerContainers/buildingModel.tsx:144]:
```suggestion
  const defines: Record<string, string> = useMemo(() => {
    hiddenFeaturesIds: JSON.stringify(hiddenFeatures),
  }, []);
```

File: extension/src/shared/layerContainers/buildingModel.tsx [hunk 3]
```diff
@@ -134,32 +133,32 @@ export const BuildingModelLayerContainer: FC<TilesetContainerProps> = ({
...
+
+  if (searchedFeatures?.onlyShow) {
+    defines.searchedFeaturesIds = JSON.stringify(searchedFeatures.features);
+  }
```
Comment (keiya01) [inline extension/src/shared/layerContainers/buildingModel.tsx:148]:
```suggestion
  const defines: Record<string, string> = useMemo(() => ({
    hiddenFeaturesIds: JSON.stringify(hiddenFeatures),
    ...(searchedFeatures?.onlyShow ? { searchedFeaturesIds: JSON.stringify(searchedFeatures.features) } : {}),
  }, []);
```

File: extension/src/shared/reearth/hooks/useLayer.ts [hunk 4]
```diff
@@ -30,6 +32,7 @@ export const useLayer = ({
       type: "simple",
       data: data,
       events,
+      defines: defines,
```
Comment (keiya01) [inline extension/src/shared/reearth/hooks/useLayer.ts:35]: You need to pass this in [here](https://github.com/eukarya-inc/PLATEAU-VIEW-3.0/pull/172/files#diff-d7246fd453668bec2c79394f62632e08671fbbb796536d152f8f7e71b835b797R54) as well.

File: extension/src/shared/layerContainers/buildingModel.tsx (hunk 1 above)
Comment (pyshx) [reply extension/src/shared/layerContainers/buildingModel.tsx]: Ahh, Thanks for pointing that out🙇

File: extension/src/shared/reearth/hooks/useLayer.ts (hunk 4 above)
Comment (pyshx) [reply extension/src/shared/reearth/hooks/useLayer.ts:35]: Done in https://github.com/eukarya-inc/PLATEAU-VIEW-3.0/pull/172/commits/9808906e8f5f1f67760594be85a10eecf4b020bf

**Review by keiya01 — APPROVED** [review]

**Pull request merged**